### PR TITLE
Deposit: create form to search

### LINKED ISSUE #13: Deposit panel: create form to search
- Deposits should have a form to search for a tracking number

### PR DESCRIPTION
**Pull request by cdsandoval**
Close #13 

Implement a controller and views of Deposit.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -5,9 +5,19 @@ class ApplicationController < ActionController::Base
...
+  def after_sign_in_path_for(current_user)
+    if current_user.role == "admin"
+      return admin_root_path
+    elsif current_user.role == "deposit"
+      return deposit_root_path
+    else
+      return root_path
+    end
+  end
+
```
Comment (Linzeur) [inline app/controllers/application_controller.rb]: Good use of the method "after_sign_in_path_for".

File: app/controllers/deposit/home_controller.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+module Deposit
+  
+  class HomeController < ApplicationController
+
+    def report
+      authorize User, policy_class: SalesPolicy
```
Comment (Linzeur) [inline app/controllers/deposit/home_controller.rb]: The members of our team have to consider that they have to create a new policy to replace to "SalePolicy", and admin users can access to all pages of deposit too.

File: app/controllers/deposit/shipments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  class ShipmentsController < ApplicationController
+
+    def report
+      authorize User, policy_class: SalesPolicy
+    end
+
+    def show
+      @shipment = Shipment.find_by_tracking_id(params[:tracking_id])
+
+    unless @shipment
```
Comment (Linzeur) [inline app/controllers/deposit/shipments_controller.rb]: Be careful with the indentation

Comment (cdsandoval) [reply app/controllers/deposit/shipments_controller.rb]: You'right, thanks for it.

File: app/controllers/deposit/home_controller.rb (hunk 2 above)
Comment (cdsandoval) [reply app/controllers/deposit/home_controller.rb]: Yes, this is just a example of the other issues.

**Pull request merged**